In [1]:
import pandas as pd
import numpy as np

# CustomerDemographic

In [2]:
df = pd.read_excel("D:\\Forage\\KPMG\\KPMG_VI_New_raw_data_update_final.xlsx", sheet_name= 'CustomerDemographic')

In [3]:
df.head()

Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only.   \
0                                        customer_id                                                                                                                              
1                                                  1                                                                                                                              
2                                                  2                                                                                                                              
3                                                  3                                                                                                                              
4                                                  4                                                                                                                              

   Unnamed: 1 Unnamed: 2 Unnamed: 3                           Unnamed: 4  \
0  first_name  last_name     gender  past_3_years_bike_related_purchases   
1     Laraine  Medendorp          F                                   93   
2         Eli    Bockman       Male                                   81   
3       Arlin     Dearle       Male                                   61   
4      Talbot        NaN       Male                                   33   

            Unnamed: 5              Unnamed: 6             Unnamed: 7  \
0                  DOB               job_title  job_industry_category   
1  1953-10-12 00:00:00     Executive Secretary                 Health   
2  1980-12-16 00:00:00  Administrative Officer     Financial Services   
3  1954-01-20 00:00:00      Recruiting Manager               Property   
4  1961-10-03 00:00:00                     NaN                     IT   

       Unnamed: 8          Unnamed: 9  \
0  wealth_segment  deceased_indicator   
1   Mass Customer                   N   
2   Mass Customer                   N   
3   Mass Customer                   N   
4   Mass Customer                   N   

                                         Unnamed: 10 Unnamed: 11 Unnamed: 12  
0                                            default    owns_car      tenure  
1                                                 "'         Yes          11  
2                       <script>alert('hi')</script>         Yes          16  
3                                2018-02-01 00:00:00         Yes          15  
4  () { _; } >_[$($())] { touch /tmp/blns.shellsh...          No           7

In [4]:
df

Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only.   \
0                                           customer_id                                                                                                                              
1                                                     1                                                                                                                              
2                                                     2                                                                                                                              
3                                                     3                                                                                                                              
4                                                     4                                                                                                                              
...                                                 ...                                                                                                                              
3996                                               3996                                                                                                                              
3997                                               3997                                                                                                                              
3998                                               3998                                                                                                                              
3999                                               3999                                                                                                                              
4000                                               4000                                                                                                                              

      Unnamed: 1 Unnamed: 2 Unnamed: 3                           Unnamed: 4  \
0     first_name  last_name     gender  past_3_years_bike_related_purchases   
1        Laraine  Medendorp          F                                   93   
2            Eli    Bockman       Male                                   81   
3          Arlin     Dearle       Male                                   61   
4         Talbot        NaN       Male                                   33   
...          ...        ...        ...                                  ...   
3996     Rosalia   Halgarth     Female                                    8   
3997      Blanch     Nisuis     Female                                   87   
3998      Sarene    Woolley          U                                   60   
3999   Patrizius        NaN       Male                                   11   
4000       Kippy    Oldland       Male                                   76   

               Unnamed: 5              Unnamed: 6             Unnamed: 7  \
0                     DOB               job_title  job_industry_category   
1     1953-10-12 00:00:00     Executive Secretary                 Health   
2     1980-12-16 00:00:00  Administrative Officer     Financial Services   
3     1954-01-20 00:00:00      Recruiting Manager               Property   
4     1961-10-03 00:00:00                     NaN                     IT   
...                   ...                     ...                    ...   
3996  1975-08-09 00:00:00   VP Product Management                 Health   
3997  2001-07-13 00:00:00         Statistician II          Manufacturing   
3998                  NaN       Assistant Manager                     IT   
3999  1973-10-24 00:00:00                     NaN          Manufacturing   
4000  1991-11-05 00:00:00    Softwa

In [5]:
df.isnull().sum()

Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only.       0
Unnamed: 1                                                                                                                                                                         0
Unnamed: 2                                                                                                                                                                       125
Unnamed: 3                                                                                                                                                                         0
Unnamed: 4                                                                                                                                                                         0
Unnamed: 5                                                                                     

In [6]:
df.columns

Index(['Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only. ',
       'Unnamed: 1', 'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5',
       'Unnamed: 6', 'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10',
       'Unnamed: 11', 'Unnamed: 12'],
      dtype='object')

In [7]:
df = df.rename(columns={'Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only. ' :'customer_id',
       'Unnamed: 1':'first_name', 'Unnamed: 2':'last_name', 'Unnamed: 3':'gender', 'Unnamed: 4':'purchases', 'Unnamed: 5':'DOB',
       'Unnamed: 6':'job_title', 'Unnamed: 7':'job_industry_category', 'Unnamed: 8':'wealth_segment', 'Unnamed: 9':'deceased_indicator', 'Unnamed: 10':'default',
       'Unnamed: 11':'owns_car', 'Unnamed: 12':'tenure'})

In [8]:
df.head()

customer_id  first_name  last_name  gender  \
0  customer_id  first_name  last_name  gender   
1            1     Laraine  Medendorp       F   
2            2         Eli    Bockman    Male   
3            3       Arlin     Dearle    Male   
4            4      Talbot        NaN    Male   

                             purchases                  DOB  \
0  past_3_years_bike_related_purchases                  DOB   
1                                   93  1953-10-12 00:00:00   
2                                   81  1980-12-16 00:00:00   
3                                   61  1954-01-20 00:00:00   
4                                   33  1961-10-03 00:00:00   

                job_title  job_industry_category  wealth_segment  \
0               job_title  job_industry_category  wealth_segment   
1     Executive Secretary                 Health   Mass Customer   
2  Administrative Officer     Financial Services   Mass Customer   
3      Recruiting Manager               Property   Mass Customer   
4                     NaN                     IT   Mass Customer   

   deceased_indicator                                            default  \
0  deceased_indicator                                            default   
1                   N                                                 "'   
2                   N                       <script>alert('hi')</script>   
3                   N                                2018-02-01 00:00:00   
4                   N  () { _; } >_[$($())] { touch /tmp/blns.shellsh...   

   owns_car  tenure  
0  owns_car  tenure  
1       Yes      11  
2       Yes      16  
3       Yes      15  
4        No       7

In [9]:
df.iloc[0,0:]

customer_id                                      customer_id
first_name                                        first_name
last_name                                          last_name
gender                                                gender
purchases                past_3_years_bike_related_purchases
DOB                                                      DOB
job_title                                          job_title
job_industry_category                  job_industry_category
wealth_segment                                wealth_segment
deceased_indicator                        deceased_indicator
default                                              default
owns_car                                            owns_car
tenure                                                tenure
Name: 0, dtype: object

In [10]:
df = df.drop(index=0)

In [11]:
df.head()

customer_id      first_name  last_name  gender purchases  \
1           1         Laraine  Medendorp       F        93   
2           2             Eli    Bockman    Male        81   
3           3           Arlin     Dearle    Male        61   
4           4          Talbot        NaN    Male        33   
5           5  Sheila-kathryn     Calton  Female        56   

                   DOB               job_title job_industry_category  \
1  1953-10-12 00:00:00     Executive Secretary                Health   
2  1980-12-16 00:00:00  Administrative Officer    Financial Services   
3  1954-01-20 00:00:00      Recruiting Manager              Property   
4  1961-10-03 00:00:00                     NaN                    IT   
5  1977-05-13 00:00:00           Senior Editor                   NaN   

      wealth_segment deceased_indicator  \
1      Mass Customer                  N   
2      Mass Customer                  N   
3      Mass Customer                  N   
4      Mass Customer                  N   
5  Affluent Customer                  N   

                                             default owns_car tenure  
1                                                 "'      Yes     11  
2                       <script>alert('hi')</script>      Yes     16  
3                                2018-02-01 00:00:00      Yes     15  
4  () { _; } >_[$($())] { touch /tmp/blns.shellsh...       No      7  
5                                                NIL      Yes      8

In [12]:
df['DOB'] = pd.to_datetime(df['DOB'])

In [13]:
df.isnull().sum()

customer_id                0
first_name                 0
last_name                125
gender                     0
purchases                  0
DOB                       87
job_title                506
job_industry_category    656
wealth_segment             0
deceased_indicator         0
default                  302
owns_car                   0
tenure                    87
dtype: int64

In [14]:
df.isna().sum()

customer_id                0
first_name                 0
last_name                125
gender                     0
purchases                  0
DOB                       87
job_title                506
job_industry_category    656
wealth_segment             0
deceased_indicator         0
default                  302
owns_car                   0
tenure                    87
dtype: int64

In [15]:
df.duplicated().sum()

0

In [16]:
df

customer_id      first_name  last_name  gender purchases        DOB  \
1              1         Laraine  Medendorp       F        93 1953-10-12   
2              2             Eli    Bockman    Male        81 1980-12-16   
3              3           Arlin     Dearle    Male        61 1954-01-20   
4              4          Talbot        NaN    Male        33 1961-10-03   
5              5  Sheila-kathryn     Calton  Female        56 1977-05-13   
...          ...             ...        ...     ...       ...        ...   
3996        3996         Rosalia   Halgarth  Female         8 1975-08-09   
3997        3997          Blanch     Nisuis  Female        87 2001-07-13   
3998        3998          Sarene    Woolley       U        60        NaT   
3999        3999       Patrizius        NaN    Male        11 1973-10-24   
4000        4000           Kippy    Oldland    Male        76 1991-11-05   

                   job_title job_industry_category     wealth_segment  \
1        Executive Secretary                Health      Mass Customer   
2     Administrative Officer    Financial Services      Mass Customer   
3         Recruiting Manager              Property      Mass Customer   
4                        NaN                    IT      Mass Customer   
5              Senior Editor                   NaN  Affluent Customer   
...                      ...                   ...                ...   
3996   VP Product Management                Health      Mass Customer   
3997         Statistician II         Manufacturing     High Net Worth   
3998       Assistant Manager                    IT     High Net Worth   
3999                     NaN         Manufacturing  Affluent Customer   
4000    Software Engineer IV                   NaN  Affluent Customer   

     deceased_indicator                                            default  \
1                     N                                                 "'   
2                     N                       <script>alert('hi')</script>   
3                     N                                2018-02-01 00:00:00   
4                     N  () { _; } >_[$($())] { touch /tmp/blns.shellsh...   
5                     N                                                NIL   
...                 ...                                                ...   
3996                  N                                               -100   
3997                  N                                           â¦test⧠  
3998                  N                                                NaN   
3999                  N                             ¡â¢Â£Â¢â§¶â¢ÂªÂºââ    
4000                  N                                                0/0   

     owns_car tenure  
1         Yes     11  
2         Yes     16  
3         Yes     15  
4          No      7  
5         Yes      8  
...       ...    ...  
3996       No     19  
3997      Yes      1  
3998       No    NaN  
3999      Yes     10  
4000       No     11  

[4000 rows x 13 columns]

In [17]:
125/len(df)*100

3.125

In [18]:
print("Percentage of blank columns, na,null values by column")

print('last_name :' ,round(125/len(df)*100,2))
print('DOB :'  ,87/len(df)*100)
print('job_title :',round(506/len(df)*100,2))
print('job_industry_category :' , round(656/len(df)*100,2))
print('default :' ,302/len(df)*100)
print('tenure :' ,round(87/len(df)*100,2))

Percentage of blank columns, na,null values by column
last_name : 3.12
DOB : 2.175
job_title : 12.65
job_industry_category : 16.4
default : 7.55
tenure : 2.17


In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4000 entries, 1 to 4000
Data columns (total 13 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   customer_id            4000 non-null   object        
 1   first_name             4000 non-null   object        
 2   last_name              3875 non-null   object        
 3   gender                 4000 non-null   object        
 4   purchases              4000 non-null   object        
 5   DOB                    3913 non-null   datetime64[ns]
 6   job_title              3494 non-null   object        
 7   job_industry_category  3344 non-null   object        
 8   wealth_segment         4000 non-null   object        
 9   deceased_indicator     4000 non-null   object        
 10  default                3698 non-null   object        
 11  owns_car               4000 non-null   object        
 12  tenure                 3913 non-null   object        
dtypes: 

In [20]:
df = df.sort_values(by='DOB').reset_index()

In [21]:
df.head()

index customer_id first_name last_name  gender purchases        DOB  \
0     34          34   Jephthah  Bachmann       U        59 1843-12-21   
1    720         720     Darrel     Canet    Male        67 1931-10-23   
2   1092        1092     Katlin   Creddon  Female        56 1935-08-22   
3   3410        3410    Merrili   Brittin  Female        93 1940-09-22   
4   2413        2413      Abbey    Murrow    Male        27 1943-08-11   

                  job_title job_industry_category     wealth_segment  \
0           Legal Assistant                    IT  Affluent Customer   
1        Recruiting Manager                Retail  Affluent Customer   
2        VP Quality Control                Retail      Mass Customer   
3                       NaN              Property      Mass Customer   
4  Environmental Specialist         Manufacturing     High Net Worth   

  deceased_indicator             default owns_car tenure  
0                  N                 NaN       No     20  
1                  N  1;DROP TABLE users       No      6  
2                  N  ì¬íê³¼íì ì´íì°êµ¬ì       No      5  
3                  N                   á       No     16  
4                  N                  á       Yes     17

In [22]:
print("Min =", min(df['DOB']))
print("Max =", max(df['DOB']))

Min = 1843-12-21 00:00:00
Max = 2002-03-11 00:00:00


### If you can see above clearly that there is year 1843 is mentioned which means the person is 179 years old 

In [23]:
df.nunique()

index                    4000
customer_id              4000
first_name               3139
last_name                3725
gender                      6
purchases                 100
DOB                      3448
job_title                 195
job_industry_category       9
wealth_segment              3
deceased_indicator          2
default                    90
owns_car                    2
tenure                     22
dtype: int64

# Transactions

In [24]:
df1 = pd.read_excel("D:\\Forage\\KPMG\\KPMG_VI_New_raw_data_update_final.xlsx", sheet_name='Transactions')

In [25]:
df1.head()

Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only.   \
0                                     transaction_id                                                                                                                              
1                                                  1                                                                                                                              
2                                                  2                                                                                                                              
3                                                  3                                                                                                                              
4                                                  4                                                                                                                              

   Unnamed: 1   Unnamed: 2           Unnamed: 3    Unnamed: 4    Unnamed: 5  \
0  product_id  customer_id     transaction_date  online_order  order_status   
1           2         2950  2017-02-25 00:00:00         False      Approved   
2           3         3120  2017-05-21 00:00:00          True      Approved   
3          37          402  2017-10-16 00:00:00         False      Approved   
4          88         3135  2017-08-31 00:00:00         False      Approved   

       Unnamed: 6    Unnamed: 7     Unnamed: 8    Unnamed: 9 Unnamed: 10  \
0           brand  product_line  product_class  product_size  list_price   
1           Solex      Standard         medium        medium       71.49   
2   Trek Bicycles      Standard         medium         large     2091.47   
3      OHM Cycles      Standard            low        medium     1793.43   
4  Norco Bicycles      Standard         medium        medium     1198.46   

     Unnamed: 11              Unnamed: 12  
0  standard_cost  product_first_sold_date  
1          53.62                    41245  
2         388.92                    41701  
3         248.82                    36361  
4          381.1                    36145

In [26]:
df1.columns

Index(['Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only. ',
       'Unnamed: 1', 'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5',
       'Unnamed: 6', 'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10',
       'Unnamed: 11', 'Unnamed: 12'],
      dtype='object')

In [27]:
#transaction_id	product_id	customer_id	transaction_date	online_order	order_status	brand	product_line	product_class	product_size	list_price	standard_cost	product_first_sold_date

In [28]:
df1 = df1.rename(columns={'Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only. ':'transaction_id',
       'Unnamed: 1':'product_id', 'Unnamed: 2':'customer_id', 'Unnamed: 3':'transaction_date', 'Unnamed: 4':'online_order', 'Unnamed: 5':'order_status',
       'Unnamed: 6':'brand', 'Unnamed: 7':'product_line', 'Unnamed: 8':'product_class', 'Unnamed: 9':'product_size', 'Unnamed: 10':'list_price',
       'Unnamed: 11':'standard_cost', 'Unnamed: 12':'product_first_sold_date'})

In [29]:
df1.head()

transaction_id  product_id  customer_id     transaction_date  online_order  \
0  transaction_id  product_id  customer_id     transaction_date  online_order   
1               1           2         2950  2017-02-25 00:00:00         False   
2               2           3         3120  2017-05-21 00:00:00          True   
3               3          37          402  2017-10-16 00:00:00         False   
4               4          88         3135  2017-08-31 00:00:00         False   

   order_status           brand  product_line  product_class  product_size  \
0  order_status           brand  product_line  product_class  product_size   
1      Approved           Solex      Standard         medium        medium   
2      Approved   Trek Bicycles      Standard         medium         large   
3      Approved      OHM Cycles      Standard            low        medium   
4      Approved  Norco Bicycles      Standard         medium        medium   

   list_price  standard_cost  product_first_sold_date  
0  list_price  standard_cost  product_first_sold_date  
1       71.49          53.62                    41245  
2     2091.47         388.92                    41701  
3     1793.43         248.82                    36361  
4     1198.46          381.1                    36145

In [30]:
df1 = df1.drop(index=0)

In [31]:
df1.head()

transaction_id product_id customer_id     transaction_date online_order  \
1              1          2        2950  2017-02-25 00:00:00        False   
2              2          3        3120  2017-05-21 00:00:00         True   
3              3         37         402  2017-10-16 00:00:00        False   
4              4         88        3135  2017-08-31 00:00:00        False   
5              5         78         787  2017-10-01 00:00:00         True   

  order_status           brand product_line product_class product_size  \
1     Approved           Solex     Standard        medium       medium   
2     Approved   Trek Bicycles     Standard        medium        large   
3     Approved      OHM Cycles     Standard           low       medium   
4     Approved  Norco Bicycles     Standard        medium       medium   
5     Approved  Giant Bicycles     Standard        medium        large   

  list_price standard_cost product_first_sold_date  
1      71.49         53.62                   41245  
2    2091.47        388.92                   41701  
3    1793.43        248.82                   36361  
4    1198.46         381.1                   36145  
5     1765.3        709.48                   42226

In [32]:
df1['transaction_date'] = pd.to_datetime(df1['transaction_date'])

In [33]:
df1.head()

transaction_id product_id customer_id transaction_date online_order  \
1              1          2        2950       2017-02-25        False   
2              2          3        3120       2017-05-21         True   
3              3         37         402       2017-10-16        False   
4              4         88        3135       2017-08-31        False   
5              5         78         787       2017-10-01         True   

  order_status           brand product_line product_class product_size  \
1     Approved           Solex     Standard        medium       medium   
2     Approved   Trek Bicycles     Standard        medium        large   
3     Approved      OHM Cycles     Standard           low       medium   
4     Approved  Norco Bicycles     Standard        medium       medium   
5     Approved  Giant Bicycles     Standard        medium        large   

  list_price standard_cost product_first_sold_date  
1      71.49         53.62                   41245  
2    2091.47        388.92                   41701  
3    1793.43        248.82                   36361  
4    1198.46         381.1                   36145  
5     1765.3        709.48                   42226

In [34]:
df1.shape

(20000, 13)

In [35]:
df1.isnull().sum()

transaction_id               0
product_id                   0
customer_id                  0
transaction_date             0
online_order               360
order_status                 0
brand                      197
product_line               197
product_class              197
product_size               197
list_price                   0
standard_cost              197
product_first_sold_date    197
dtype: int64

In [36]:
df1.isna().sum()

transaction_id               0
product_id                   0
customer_id                  0
transaction_date             0
online_order               360
order_status                 0
brand                      197
product_line               197
product_class              197
product_size               197
list_price                   0
standard_cost              197
product_first_sold_date    197
dtype: int64

In [37]:
print('online_order :' ,round(360/len(df1)*100,2))
print("brand ,product_line,product_class,product_size,standard_cost,product_first_sold_date :" ,round(197/len(df1)*100,2))

online_order : 1.8
brand ,product_line,product_class,product_size,standard_cost,product_first_sold_date : 0.98


In [38]:
df1.duplicated().sum()

0

In [39]:
print("Min =", min(df1['transaction_date']))
print("Max =", max(df1['transaction_date']))

Min = 2017-01-01 00:00:00
Max = 2017-12-30 00:00:00


In [40]:
df2 = pd.read_excel("D:\\Forage\\KPMG\\KPMG_VI_New_raw_data_update_final.xlsx", sheet_name='CustomerAddress')

In [41]:
df2.head()

Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only.   \
0                                        customer_id                                                                                                                              
1                                                  1                                                                                                                              
2                                                  2                                                                                                                              
3                                                  4                                                                                                                              
4                                                  5                                                                                                                              

            Unnamed: 1 Unnamed: 2       Unnamed: 3 Unnamed: 4  \
0              address   postcode            state    country   
1   060 Morning Avenue       2016  New South Wales  Australia   
2  6 Meadow Vale Court       2153  New South Wales  Australia   
3   0 Holy Cross Court       4211              QLD  Australia   
4  17979 Del Mar Point       2448  New South Wales  Australia   

           Unnamed: 5  
0  property_valuation  
1                  10  
2                  10  
3                   9  
4                   4

In [42]:
df2.columns

Index(['Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only. ',
       'Unnamed: 1', 'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5'],
      dtype='object')

In [43]:
#customer_id	address	postcode	state	country	property_valuation

In [44]:
df2 = df2.rename(columns={'Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only. ':'customer_id',
       'Unnamed: 1':'address', 'Unnamed: 2':'postcode', 'Unnamed: 3':'state', 'Unnamed: 4':'country', 'Unnamed: 5':'property_valuation'})

In [45]:
df2.head()

customer_id              address  postcode            state    country  \
0  customer_id              address  postcode            state    country   
1            1   060 Morning Avenue      2016  New South Wales  Australia   
2            2  6 Meadow Vale Court      2153  New South Wales  Australia   
3            4   0 Holy Cross Court      4211              QLD  Australia   
4            5  17979 Del Mar Point      2448  New South Wales  Australia   

   property_valuation  
0  property_valuation  
1                  10  
2                  10  
3                   9  
4                   4

In [46]:
df2.isnull().sum()

customer_id           0
address               0
postcode              0
state                 0
country               0
property_valuation    0
dtype: int64

In [47]:
df2.isna().sum()

customer_id           0
address               0
postcode              0
state                 0
country               0
property_valuation    0
dtype: int64

In [48]:
df2.duplicated().sum()

0

In [49]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4000 entries, 0 to 3999
Data columns (total 6 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   customer_id         4000 non-null   object
 1   address             4000 non-null   object
 2   postcode            4000 non-null   object
 3   state               4000 non-null   object
 4   country             4000 non-null   object
 5   property_valuation  4000 non-null   object
dtypes: object(6)
memory usage: 187.6+ KB


In [62]:
df3 = pd.read_excel("D:\\Forage\\KPMG\\KPMG_VI_New_raw_data_update_final.xlsx", sheet_name='NewCustomerList')

In [63]:
df3.head()

Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only.   \
0                                         first_name                                                                                                                              
1                                            Chickie                                                                                                                              
2                                              Morly                                                                                                                              
3                                            Ardelis                                                                                                                              
4                                             Lucine                                                                                                                              

  Unnamed: 1 Unnamed: 2                           Unnamed: 3  \
0  last_name     gender  past_3_years_bike_related_purchases   
1    Brister       Male                                   86   
2     Genery       Male                                   69   
3  Forrester     Female                                   10   
4      Stutt     Female                                   64   

            Unnamed: 4                  Unnamed: 5             Unnamed: 6  \
0                  DOB                   job_title  job_industry_category   
1           1957-07-12             General Manager          Manufacturing   
2           1970-03-22         Structural Engineer               Property   
3  1974-08-28 00:00:00      Senior Cost Accountant     Financial Services   
4           1979-01-28  Account Representative III          Manufacturing   

          Unnamed: 7          Unnamed: 8 Unnamed: 9  ... Unnamed: 12  \
0     wealth_segment  deceased_indicator   owns_car  ...    postcode   
1      Mass Customer                   N        Yes  ...        4500   
2      Mass Customer                   N         No  ...        2113   
3  Affluent Customer                   N         No  ...        3505   
4  Affluent Customer                   N        Yes  ...        4814   

  Unnamed: 13 Unnamed: 14         Unnamed: 15 Unnamed: 16 Unnamed: 17  \
0       state     country  property_valuation           A           B   
1         QLD   Australia                   6         0.7       0.875   
2         NSW   Australia                  11        0.87        0.87   
3         VIC   Australia                   5        1.07        1.07   
4         QLD   Australia                   1           1        1.25   

  Unnamed: 18 Unnamed: 19 Unnamed: 20 Unnamed: 21  
0           C           D        Rank       Value  
1     1.09375    0.929688           1     1.71875  
2      1.0875    0.924375           1     1.71875  
3        1.07        1.07           1     1.71875  
4        1.25        1.25           4    1.703125  

[5 rows x 22 columns]

In [66]:
df3.columns

Index(['Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only. ',
       'Unnamed: 1', 'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5',
       'Unnamed: 6', 'Unnamed: 7', 'Unnamed: 8', 'Unnamed: 9', 'Unnamed: 10',
       'Unnamed: 11', 'Unnamed: 12', 'Unnamed: 13', 'Unnamed: 14',
       'Unnamed: 15', 'Unnamed: 16', 'Unnamed: 17', 'Unnamed: 18',
       'Unnamed: 19', 'Unnamed: 20', 'Unnamed: 21'],
      dtype='object')

In [67]:
df3 = df3.rename(columns={'Note: The data and information in this document is reflective of a hypothetical situation and client. This document is to be used for KPMG Virtual Internship purposes only. ':'first_name',
       'Unnamed: 1':'last_name', 'Unnamed: 2':'gender', 'Unnamed: 3':'past_3_years_bike_related_purchases', 'Unnamed: 4':'DOB', 'Unnamed: 5':'job_title',
       'Unnamed: 6':'job_industry_category', 'Unnamed: 7':'wealth_segment', 'Unnamed: 8':'deceased_indicator', 'Unnamed: 9':'owns_car', 'Unnamed: 10':'tenure',
       'Unnamed: 11':'address', 'Unnamed: 12':'postcode', 'Unnamed: 13':'state', 'Unnamed: 14':'country',
       'Unnamed: 15':'property_valuation', 'Unnamed: 16':'A', 'Unnamed: 17':'B', 'Unnamed: 18':'C',
       'Unnamed: 19':'D', 'Unnamed: 20':'Rank', 'Unnamed: 21':'Value'})

In [68]:
df3.head()

first_name  last_name  gender  past_3_years_bike_related_purchases  \
0  first_name  last_name  gender  past_3_years_bike_related_purchases   
1     Chickie    Brister    Male                                   86   
2       Morly     Genery    Male                                   69   
3     Ardelis  Forrester  Female                                   10   
4      Lucine      Stutt  Female                                   64   

                   DOB                   job_title  job_industry_category  \
0                  DOB                   job_title  job_industry_category   
1           1957-07-12             General Manager          Manufacturing   
2           1970-03-22         Structural Engineer               Property   
3  1974-08-28 00:00:00      Senior Cost Accountant     Financial Services   
4           1979-01-28  Account Representative III          Manufacturing   

      wealth_segment  deceased_indicator  owns_car  ...  postcode  state  \
0     wealth_segment  deceased_indicator  owns_car  ...  postcode  state   
1      Mass Customer                   N       Yes  ...      4500    QLD   
2      Mass Customer                   N        No  ...      2113    NSW   
3  Affluent Customer                   N        No  ...      3505    VIC   
4  Affluent Customer                   N       Yes  ...      4814    QLD   

     country  property_valuation     A      B        C         D  Rank  \
0    country  property_valuation     A      B        C         D  Rank   
1  Australia                   6   0.7  0.875  1.09375  0.929688     1   
2  Australia                  11  0.87   0.87   1.0875  0.924375     1   
3  Australia                   5  1.07   1.07     1.07      1.07     1   
4  Australia                   1     1   1.25     1.25      1.25     4   

      Value  
0     Value  
1   1.71875  
2   1.71875  
3   1.71875  
4  1.703125  

[5 rows x 22 columns]

In [70]:
df3 = df3.drop(index=0)

In [71]:
df3.duplicated().sum()

0

In [72]:
df3.isnull().sum()

first_name                               0
last_name                               29
gender                                   0
past_3_years_bike_related_purchases      0
DOB                                     17
job_title                              106
job_industry_category                  165
wealth_segment                           0
deceased_indicator                       0
owns_car                                 0
tenure                                   0
address                                  0
postcode                                 0
state                                    0
country                                  0
property_valuation                       0
A                                        0
B                                        0
C                                        0
D                                        0
Rank                                     0
Value                                    0
dtype: int64

In [73]:
df3.isnull().sum()/len(df3)*100

first_name                              0.0
last_name                               2.9
gender                                  0.0
past_3_years_bike_related_purchases     0.0
DOB                                     1.7
job_title                              10.6
job_industry_category                  16.5
wealth_segment                          0.0
deceased_indicator                      0.0
owns_car                                0.0
tenure                                  0.0
address                                 0.0
postcode                                0.0
state                                   0.0
country                                 0.0
property_valuation                      0.0
A                                       0.0
B                                       0.0
C                                       0.0
D                                       0.0
Rank                                    0.0
Value                                   0.0
dtype: float64